alfresco-transform-imagemagick/alfresco-transform-imagemagick-boot/src/test/java/org/alfresco/transformer/ImageMagickControllerTest.java

```diff
@@ -26,77 +26,23 @@
-import static org.mockito.Mockito.when;
-import static org.springframework.http.HttpHeaders.ACCEPT;
-import static org.springframework.http.HttpHeaders.CONTENT_DISPOSITION;
-import static org.springframework.http.HttpHeaders.CONTENT_TYPE;
-import static org.springframework.http.HttpStatus.BAD_REQUEST;
-import static org.springframework.http.HttpStatus.CREATED;
-import static org.springframework.http.HttpStatus.OK;
-import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
-import static org.springframework.http.MediaType.IMAGE_PNG_VALUE;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.header;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
-import static org.springframework.util.StringUtils.getFilenameExtension;
-
-import java.io.File;
 import java.io.IOException;
-import java.nio.file.Files;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.UUID;
 
-import org.alfresco.transform.client.model.TransformReply;
-import org.alfresco.transform.client.model.TransformRequest;
-import org.alfresco.transformer.executors.ImageMagickCommandExecutor;
-import org.alfresco.transformer.executors.RuntimeExec;
-import org.alfresco.transformer.model.FileRefEntity;
-import org.alfresco.transformer.model.FileRefResponse;
 import org.junit.Before;
-import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.mockito.Mock;
-import org.mockito.stubbing.Answer;
 import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
 import org.springframework.boot.test.mock.mockito.SpyBean;
-import org.springframework.core.io.FileSystemResource;
-import org.springframework.core.io.Resource;
-import org.springframework.http.HttpHeaders;
-import org.springframework.http.ResponseEntity;
-import org.springframework.mock.web.MockMultipartFile;
 import org.springframework.test.context.junit4.SpringRunner;
 import org.springframework.test.util.ReflectionTestUtils;
-import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
 
 /**
  * Test the ImageMagickController without a server.
  * Super class includes tests for the AbstractTransformerController.
  */
 @RunWith(SpringRunner.class)
 @WebMvcTest(ImageMagickController.class)
-public class ImageMagickControllerTest extends AbstractTransformerControllerTest
+public class ImageMagickControllerTest extends ImageMagickControllerTestBase
 {
-    private static final String ENGINE_CONFIG_NAME = "imagemagick_engine_config.json";
-
-    @Mock
-    private ExecutionResult mockExecutionResult;
-
-    @Mock
-    private RuntimeExec mockTransformCommand;
-
-    @Mock
-    private RuntimeExec mockCheckCommand;
-
-    private ImageMagickCommandExecutor commandExecutor = new ImageMagickCommandExecutor();
 
     @SpyBean
     private ImageMagickController controller;
```

Comment:
Separating the ImageMagick tests were needed as this inherited variable could not be instantiated by AIO test classes.

Reply:
Maybe add this reason as a comment to code ?

Reply:
Added comment